src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -145,7 +212,51 @@ private void UpdateRestoreBar(bool packagesMissing)
 
         private void OnRestoreLinkClick(object sender, RoutedEventArgs e)
         {
-            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => UIRestorePackagesAsync(CancellationToken.None)).PostOnFailure(nameof(PackageRestoreBar));
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() =>
+            {
+                if (_projectContextInfo?.ProjectStyle == ProjectModel.ProjectStyle.PackageReference && _isAssetsFileMissing)
+                {
+                    return UIRestoreProjectAsync(CancellationToken.None);
+                }
+
+                return UIRestorePackagesAsync(CancellationToken.None);
+            }).PostOnFailure(nameof(PackageRestoreBar));
+        }
+
+        private async Task<bool> UIRestoreProjectAsync(CancellationToken token)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            ShowProgressUI();
+            OperationId = Guid.NewGuid();
+
+            try
+            {
+                _packageRestoreManager.PackageRestoreFailedEvent += PackageRestoreFailedEvent;
+
+                string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(token);
```

Comment:
You don't need this for the same reason.

Reply:
ScheduleRestoreAsync is also guaranteed to never throw so you don't really need the catch.